java/client/src/main/java/glide/api/commands/BaseCommands.java

```diff
@@ -19,31 +18,37 @@ static BaseCommandResponseResolver applyBaseCommandResponseResolver() {
 
   /**
    * Extracts the response from the Protobuf response and either throws an exception or returns the
-   * appropriate response has an Object
+   * appropriate response as an Object
    *
    * @param response Redis protobuf message
    * @return Response Object
    */
   static Object handleObjectResponse(Response response) {
     // return function to convert protobuf.Response into the response object by
     // calling valueFromPointer
-    return BaseCommands.applyBaseCommandResponseResolver().apply(response);
+    return applyBaseCommandResponseResolver().apply(response);
   }
 
-  public static List<Object> handleTransactionResponse(Response response) {
+  static Object[] handleTransactionResponse(Response response) {
     // return function to convert protobuf.Response into the response object by
     // calling valueFromPointer
 
-    List<Object> transactionResponse =
-        (List<Object>) BaseCommands.applyBaseCommandResponseResolver().apply(response);
-    return transactionResponse;
+    return (Object[]) applyBaseCommandResponseResolver().apply(response);
   }
 
   /**
-   * Execute a @see{Command} by sending command via socket manager
+   * Execute a custom {@link Command}.
    *
    * @param args arguments for the custom command
    * @return a CompletableFuture with response result from Redis
    */
   CompletableFuture<Object> customCommand(String[] args);
+
+  /**
+   * Execute a transaction of custom {@link Command}s.
+   *
+   * @param args arguments for the custom command
+   * @return a CompletableFuture with response result from Redis
+   */
+  CompletableFuture<Object[]> customTransaction(String[][] args);
```

Comment:
what about exposing? 
```
CompletableFuture<Object[]> customTransaction(Transaction transaction);
```
This is what the node client does: https://github.com/Bit-Quill/glide-for-redis/blob/46f831c5d2b31a667f7d7c53035b99e8b3851f26/node/src/RedisClient.ts#L98

Reply:
We may have to create a Transactions class that implements BaseCommands that adds the transaction calls to the list. 
Maybe also a ClusterTransactions class that implements BaseClusterCommands too... 

reference: https://github.com/Bit-Quill/glide-for-redis/blob/main/node/src/Transaction.ts#L72